sensors/Dht.py

```diff
@@ -0,0 +1,98 @@
+# inheritance library
+# import abc
+import Adafruit_DHT
+# GPIO lib
+import RPi.GPIO as GPIO           # import RPi.GPIO module
+
+from sensors.SensorException import SensorReadError
+
+
+# SensorInterface(Top Hierarchy)
+# Author     : clintonbf
+# Repository :   https://github.com/clintonbf/sensor_interface/blob/master/sensors/SensorInterface.py
+from sensors.Sensor import Sensor
+
+# unit conversion function
+# Author     : clintonbf
+# Repository : https://github.com/clintonbf/sensor_interface/tree/master/unit_conversion
+from utils.unit_conversion.temperature import celsius_to_kelvin
+
+DATA_INDICES = {
+    "relative_humidity": 0,
+        "temperature": 1,
+    }
+
+class Dht(Sensor):
+    """
+        Credit for connect_to_port and get_raw_data goes to
+        Noah MacRitchie (noah21mac@gmail.com) and Andrey Goryelov (andrey.goryelov@gmail.com)
+    """
+
+    # TODO: what is timeout ?
+    # TODO: It is just a GPIO data in, not serial connection.
+    # TODO: pass it in like : gpio_in = 1, named params
+   
+    def __init__(self, uid: int, gpio_in: str, model : str ) -> None:
+        # super().__init__()
+        self.__uid = uid
+        self.__gpio_in = gpio_in
+        # self.__timeout = timeout
+        if model.lower() == 'dht11':
+            self.__model = 11
+        # elif model.lower() == 'am2302':
+        #     self.__model = 22
+        else:
+            self.__model = 22
+        self.__reading = () # TODO meaning?
+
+    @classmethod
+    def __subclasshook__(cls, subclass):
+        return(hasattr(subclass, 'format_data') and
+               callable(subclass.format_data) or
+               NotImplemented)
+
+    # Sets the GPIO to be in input mode
+    def connect_to_sensor(self) -> int:
```

Comment:
The method signature for connect_to_sensor() has no return value. Update to reflect that (or make an argument for updating the signature)

Reply:
I will remove the return int syntax to match the signature in Sensor.py.

The signaturue for connect_to_sensor() defined in the Sensor interface does not have a return value, but I saw you were returning a serial.Serial() object with the variable name of 'ser' in your code. I did not know you can do that in python.

Reply:
I did some digging and returning here is **ok** 

I revoke my initial request to remove the return type.